third_party/terraform/utils/config.go.erb

```diff
@@ -164,162 +234,206 @@ func (c *Config) LoadAndValidate() error {
 	c.clientCompute.UserAgent = userAgent
 	c.clientCompute.BasePath = computeClientBasePath
 
-	log.Printf("[INFO] Instantiating GCE Beta client...")
+	computeBetaClientBasePath := removeBasePathVersion(c.ComputeBetaBasePath) + "beta/projects/"
+	log.Printf("[INFO] Instantiating GCE Beta client for path %s", computeBetaClientBasePath)
 	c.clientComputeBeta, err = computeBeta.NewService(context, option.WithHTTPClient(client))
 	if err != nil {
 		return err
 	}
 	c.clientComputeBeta.UserAgent = userAgent
+	c.clientComputeBeta.BasePath = computeBetaClientBasePath
 
-	log.Printf("[INFO] Instantiating GKE client...")
+	containerClientBasePath := removeBasePathVersion(c.ContainerBasePath)
+	log.Printf("[INFO] Instantiating GKE client for path %s", containerClientBasePath)
 	c.clientContainer, err = container.NewService(context, option.WithHTTPClient(client))
 	if err != nil {
 		return err
 	}
 	c.clientContainer.UserAgent = userAgent
+	c.clientContainer.BasePath = containerClientBasePath
 
-	log.Printf("[INFO] Instantiating GKE Beta client...")
+	containerBetaClientBasePath := removeBasePathVersion(c.ContainerBetaBasePath)
+	log.Printf("[INFO] Instantiating GKE Beta client for path %s", containerBetaClientBasePath)
 	c.clientContainerBeta, err = containerBeta.NewService(context, option.WithHTTPClient(client))
 	if err != nil {
 		return err
 	}
 	c.clientContainerBeta.UserAgent = userAgent
+	c.clientContainerBeta.BasePath = containerBetaClientBasePath
 
-	log.Printf("[INFO] Instantiating Google Cloud DNS client...")
+	dnsClientBasePath := removeBasePathVersion(c.DnsBasePath) + "v1/projects/"
```

Comment:
How come some of these add parts of the URL onto the end and some don't?

Reply:
While most APIs use the form `https://cloudscheduler.googleapis.com` in the client libs, DNS and Compute append `{{version/projects/` and SQL appends `{{version}}/`. Since they're hardcoded values in the other client libs, I hardcoded these as well.

Reply:
(also, commented on the first occurence)